src/java/voldemort/server/protocol/admin/AdminServiceRequestHandler.java

```diff
@@ -1943,10 +1943,25 @@ private Message handleDisableStoreVersion(VAdminProto.DisableStoreVersionRequest
         }
 
         if (response.getDisableSuccess()) {
-            // Then we also want to put the server in offline mode
+            // Voldemort Client can detect and handle failure only at a server
+            // level. Disabling a store inside Server is not possible . Client
+            // creates connection to the server and shares the same connection
+            // for multiple stores. Client failure detection is based on this
+            // connection error. So entire server will be marked as offline for
+            // now.
+
+            boolean roFetchState = metadataStore.getReadOnlyFetchEnabledUnlocked();
+
             VAdminProto.SetOfflineStateRequest offlineStateRequest =
                     VAdminProto.SetOfflineStateRequest.newBuilder().setOfflineMode(true).build();
             handleSetOfflineState(offlineStateRequest);
```

Comment:
What I'm saying is: let's make the offline state consistent: always allow fetches, by modifying the logic in ```MetadataStore.setOfflineState``` so that we don't need to pass parameters through. This makes it easier to reason about what's going on, since by default the offline state means just one thing (unless manually overridden by the operator).